zookeeper/src/main/scala/knobs/Zookeeper.scala

```diff
@@ -174,11 +176,12 @@ object ZooKeeper {
    * close.run
    * ```
    */
-  def unsafeFromResource(customConfig: List[KnobsResource]): (ResourceBox, Task[Unit]) = unsafe(customConfig)
+  def unsafeFromResource(customConfig: List[KnobsResource])(implicit ec: ExecutionContext): (ResourceBox, IO[Unit]) =
+    unsafe(customConfig)
 
-  protected def unsafe(config: List[KnobsResource] = defaultCfg): (ResourceBox, Task[Unit]) = {
-    val (box, c) = doZK(config).unsafePerformSync
-    (box, Task.delay(c.close))
+  protected def unsafe(config: List[KnobsResource] = null)(implicit ec: ExecutionContext): (ResourceBox, IO[Unit]) = {
```

Comment:
Not sure what to do here - `defaultCfg` now needs `ExecutionContext`, and afaik there's no way to pass in the `ec` param because ordering, blah blah. Perhaps here we are OK with using `ExecutionContext.Implicits.global` ?

Reply:
What ordering?  I'm confused.

Reply:
To preserve the old behavior we need `defaultCfg` as the default for `config`, but in order to do that we now need to pass an `ExecutionContext` (the old behavior was relying on the sketchy default in `Strategy`'s companion object) - however afaik there is no way to pass an implicit `ExecutionContext` to `defaultCfg` since `ec` appears in a later parameter list.

Reply:
Oh, I see.  Yeah, that's ugly.  An `ExeuctionContext => List[KnobsResource]` would untangle it, but be ugly in a different way.  I don't have a solution that makes me happy here.